backend/model/Users.js

```diff
@@ -6,103 +6,222 @@ require("firebase/auth");
+  // takes in signed-in user 
+  getUserProfile: async function (current_user) {
+
+    var userDocRef = usersCollection.doc(current_user.uid);
+    var userProfile;
+
+    // access document
+    await userDocRef.get().then(function(userDoc) {
+      if (userDoc.exists) {
+        // set userProfile to retrieved data
+        // not sure this is how to retrieve data
+        userProfile = {
+          username:     userDoc.get("username"),
+          first_name:   userDoc.get("first_name"),
+          last_name:    userDoc.get("last_name"),
+          profile_pic: userDoc.get("profile_pic")
+        };
+      } else {
+        // no data under uid
+        userProfile = null;
+      }
+
+    }).catch(function(error) {
+      console.log("Error getting document: ", error);
+      userProfile = null;
+    });
+
+    return new Promise((resolve,reject) => resolve(userProfile));
+
   },
 
   // TODO: Need to fix for testing purposes
   // User and UID remains constant throughout expo session
   // In order to reset UID, close metro bundler and npm start again
   // TODO: Develop a SIGNOUT button ASAP
-  updateUserProfile: function (username, firstName, lastName) {
-    const user = firebase.auth().currentUser;
+  updateUserProfile: async function (current_user, username, firstName, lastName) {
+    var success = false;
+
+    // trim whitespace from username
     username = username.trim();
     firstName = firstName.trim();
     lastName = lastName.trim();
-    // await user.updateProfile({displayName: username});
-    users.doc(user.uid).update({
+
+    // initialize local object containing new user values
+    const user = {
+      username: username, 
       first_name: firstName,
-      last_name: lastName,
-      username: username,
-    });
-<<<<<<< Updated upstream:backend/model/Users.js
-=======
-  },
+      last_name: lastName
+    };
 
-  emailTaken: async function(email) {
-    const response = users.where('email', '==', email);
-    const query = await response.get();
-    return !query.empty;
+    // update document with data
+    await usersCollection
+      .doc(current_user.uid)
+      .update(user) // set uid document to new user values
+      .then(() => {
+        console.log(`Updated user with ID: ${current_user.uid}\n with data: ${JSON.stringify(user)}`);
+        success = true;
+      })
+      .catch((error) => {
+        console.log(`Error adding user: ${error}`);
+        success = false;
+      });
+
+      return success;
   },
 
-  usernameTaken: async function(username) {
-    console.log('user input: ', username);
-    const response = users.where('username', '==', username);
-    const query = await response.get();
-    return !query.empty;
+  uploadUserProfilePicture: async function (current_user, file) {
+    // TODO this function may not work correctly.
+    // create a cloud storage refrence
+    var storageRef = firebase
+      .storage()
+      .ref(user.uid + "/profilePicture/" + file.name);
```

Comment:
Should we make a folder in storage for users? Not sure how organization of storage is going to be yet

Reply:
I think we should. We should write somewhere what our cloud storage organization will be like